### PR TITLE
[#1769] Fix for Clojure code block

### PR DESCRIPTION
**Pull request by drbobbeaty**
There was an issue in that the rendering of the Clojure code block in
a Markdown file would not be rendered properly. This is documented in
Issue #1769 - with screen grabs. This addition to the syntax file
corrects that issue by adding the section for the Clojure code block in
the Markdown file.

### CODE REVIEW

File: Markdown/Markdown.sublime-syntax [hunk 1]
```diff
@@ -932,6 +932,21 @@ contexts:
         - match: $
           pop: true
   fenced-code-block:
+    - match: |-
+         (?x)
+          {{fenced_code_block_start}}
+          ((?i:clojure|clj))
+          \s*$\n?          # any whitespace until EOL
```
Comment (FichteFoll) [inline Markdown/Markdown.sublime-syntax]:
```suggestion
          {{fenced_code_block_trailing_infostring_characters}}
```

File: Markdown/Markdown.sublime-syntax [hunk 2]
```diff
@@ -932,6 +932,21 @@ contexts:
...
+      captures:
+        0: meta.code-fence.definition.begin.clojure.markdown-gfm
+        2: punctuation.definition.raw.code-fence.begin.markdown
+        3: constant.other.language-name.markdown
```
Comment (FichteFoll) [inline Markdown/Markdown.sublime-syntax]:
```suggestion
        5: constant.other.language-name.markdown
```

**Comment by OdatNurd** [comment]
Indeed! My fix was based on the shipped version of the syntax in the build you mentioned using, which is somewhat different from the version currently in this repo, which has unreleased changes. 😃

**Comment by drbobbeaty** [comment]
You guys are right - I should have... I'm terribly sorry about that.

**Comment by keith-hall** [comment]
Please add some syntax test assertions to ensure the changes you made work properly and won't be regressed in the future.

https://github.com/sublimehq/Packages#pull-requests
> Pull requests should:
> - Have multiple syntax tests

See https://www.sublimetext.com/docs/3/syntax.html#testing for details.
This can be done easily with https://packagecontrol.io/packages/PackageDev.

**Comment by wbond** [comment]
I'm going to hold off on this for now. The Markdown syntax grows in the number of contexts with each syntax we add. We are seeing the sanity limit hit more frequently due to the giant include graph.

**Comment by drbobbeaty** [comment]
@wbond is there something wrong with the PR that can correct this? Or is it _something else_ that is beyond this PR?

**Comment by wbond** [comment]
No, there is nothing wrong with this PR. It is more an issue of us needing to figure out one of:

 - How to try and detect which syntax may be causing the sanity limit
 - Increase the sanity limit some, profiling memory usage to see if something like 50,000 is too high
 - See if there is more significant modifications that can be made to the highlighting engine to not require compiling each external include into a syntax. This is most likely not viable, since the increase of contexts is usually due to `with_prototype` being used, creating new contexts with prepended rules.

**Comment by deathaxe** [comment]
Loading markdown files is already very slow, which makes me doubt an increased sanity limit to be a good solution in manner of performance.

**Comment by wbond** [comment]
Since build 4075 we now have embed lazy loads, so this can be merged in without a performance penalty.

**Pull request merged**